vms/platformvm/txs/fee/calculator.go

```diff
@@ -0,0 +1,142 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package fee
+
+import (
+	"time"
+
+	"github.com/ava-labs/avalanchego/utils/constants"
+	"github.com/ava-labs/avalanchego/vms/platformvm/txs"
+	"github.com/ava-labs/avalanchego/vms/platformvm/upgrade"
+)
+
+var _ txs.Visitor = (*calculator)(nil)
+
+func NewStaticCalculator(config StaticConfig, upgradeTimes upgrade.Config) Calculator {
+	return Calculator{
+		config:       config,
+		upgradeTimes: upgradeTimes,
+	}
+}
+
+type Calculator struct {
```

Comment:
Should we rename this struct `StaticFeeCalculator` or something if we're going to introduce a different dynamic fee calculator later? Or are we just going to update this struct?

Reply:
My idea was to extend a single calculator with all relevant attributes.
It may be cleaner to have two distinct `calculator`s, with their own attribute though.
Will try this in a separate PR.